### PR TITLE
Refactors Access Code

### PR DESCRIPTION
**Pull request by redmoogle**
# Document the changes in your pull request

Makes the access code more sane 
Instead of having 2 duplicate lists with maybe some extra access added
Simply just |= and only have unique accesses
Also Deduplicates some code

Implements #14610 

# Changelog

:cl:  
tweak: Access Code Overhauled
/:cl:

### CODE REVIEW

File: code/modules/jobs/job_types/_job.dm [hunk 1]
```diff
@@ -157,12 +157,10 @@
...
 
-	. = list()
+	. = src.minimal_access.Copy()
 
-	if(CONFIG_GET(flag/jobs_have_minimal_access))
-		. = src.minimal_access.Copy()
-	else
-		. = src.access.Copy()
+	if(!CONFIG_GET(flag/jobs_have_minimal_access)) // If we should give players extra access
+		. += src.access.Copy()
```
Comment (Bibby0110) [inline code/modules/jobs/job_types/_job.dm]:
```suggestion
		. |= src.access.Copy()
```

**Review by Bibby0110 — COMMENTED** [review]
access being named as it is is kinda confusing tbh. It's more like additional access

File: code/modules/jobs/job_types/_job.dm [hunk 2]
```diff
@@ -1,98 +1,83 @@
...
-	//Tells the given channels that the given mob is the new department head. See communications.dm for valid channels.
+	/// Tells the given channels that the given mob is the new department head. See communications.dm for valid channels.
 	var/list/head_announce = null
-
-	//Bitflags for the job
-	var/department_flag = NONE //Deprecated
+	// Used for something in preferences_savefile.dm
+	var/department_flag = NONE
 	var/flag = NONE //Deprecated
+	/// Automatic deadmin for a jobd. Usually head/security positions
```
Comment (adamsong) [inline code/modules/jobs/job_types/_job.dm]:
```suggestion
	/// Automatic deadmin for a job. Usually head/security positions
```

File: code/modules/jobs/job_types/cook.dm [hunk 3]
```diff
@@ -24,8 +24,8 @@
 	changed_maps = list("OmegaStation", "EclipseStation")
 
 /datum/job/cook/proc/OmegaStationChanges()
-	access = list(ACCESS_HYDROPONICS, ACCESS_BAR, ACCESS_KITCHEN, ACCESS_MORGUE)
-	minimal_access = list(ACCESS_HYDROPONICS, ACCESS_BAR, ACCESS_KITCHEN, ACCESS_MORGUE)
+	added_access = list(ACCESS_HYDROPONICS, ACCESS_BAR, ACCESS_KITCHEN, ACCESS_MORGUE)
```
Comment (adamsong) [inline code/modules/jobs/job_types/cook.dm]:
```suggestion
	added_access = list()
```

File: code/modules/jobs/job_types/head_of_security.dm [hunk 4]
```diff
@@ -12,32 +12,30 @@
...
 
 	display_order = JOB_DISPLAY_ORDER_HEAD_OF_SECURITY
+	minimal_character_age = 28
 
 	changed_maps = list("YogsPubby")
 
 /datum/job/hos/proc/YogsPubbyChanges()
-	access += ACCESS_CREMATORIUM
-	minimal_access += ACCESS_CREMATORIUM
+	added_access |= ACCESS_CREMATORIUM
```
Comment (adamsong) [inline code/modules/jobs/job_types/head_of_security.dm]: Doesn't need to be added if its base

File: code/modules/jobs/job_types/roboticist.dm [hunk 5]
```diff
@@ -14,12 +14,13 @@
...
 
-	access = list(ACCESS_ROBOTICS, ACCESS_TOX, ACCESS_TOX_STORAGE, ACCESS_TECH_STORAGE, ACCESS_MORGUE, ACCESS_MECH_SCIENCE, ACCESS_RESEARCH, ACCESS_MINERAL_STOREROOM, ACCESS_XENOBIOLOGY, ACCESS_GENETICS)
-	minimal_access = list(ACCESS_ROBOTICS, ACCESS_TECH_STORAGE, ACCESS_MORGUE, ACCESS_RESEARCH, ACCESS_MECH_SCIENCE, ACCESS_MINERAL_STOREROOM)
+	added_access = list(ACCESS_TOX, ACCESS_TOX_STORAGE, ACCESS_XENOBIOLOGY, ACCESS_GENETICS)
+	base_access = list(ACCESS_ROBOTICS, ACCESS_TECH_STORAGE, ACCESS_MORGUE, ACCESS_RESEARCH, ACCESS_MECH_SCIENCE, ACCESS_MINERAL_STOREROOM)
 	paycheck = PAYCHECK_MEDIUM
 	paycheck_department = ACCOUNT_SCI
 
 	display_order = JOB_DISPLAY_ORDER_ROBOTICIST
+	minimal_character_age = 18
```
Comment (adamsong) [inline code/modules/jobs/job_types/roboticist.dm:23]: Undocumented change, Atomize this

File: code/modules/jobs/job_types/security_officer.dm [hunk 6]
```diff
@@ -32,14 +32,14 @@
 	spawn_positions = 10
 
 /datum/job/officer/proc/YogsPubbyChanges()
-	access += ACCESS_CREMATORIUM
-	minimal_access += ACCESS_CREMATORIUM
+	added_access |= ACCESS_CREMATORIUM
```
Comment (adamsong) [inline code/modules/jobs/job_types/security_officer.dm]: Doesn't need to be added to added and base

File: code/modules/jobs/job_types/warden.dm [hunk 7]
```diff
@@ -29,8 +29,8 @@
 	changed_maps = list("YogsPubby", "OmegaStation")
 
 /datum/job/warden/proc/YogsPubbyChanges()
-	access += ACCESS_CREMATORIUM
-	minimal_access += ACCESS_CREMATORIUM
+	added_access |= ACCESS_CREMATORIUM
```
Comment (adamsong) [inline code/modules/jobs/job_types/warden.dm]: Does not need to be both added and base

File: yogstation/code/modules/jobs/job_types/network_admin.dm [hunk 8]
```diff
@@ -14,18 +14,17 @@
 
 	outfit = /datum/outfit/job/network_admin
 
-	access = list(ACCESS_TCOMSAT, ACCESS_TCOM_ADMIN, ACCESS_TECH_STORAGE, ACCESS_RC_ANNOUNCE, ACCESS_CONSTRUCTION, ACCESS_ENGINE, ACCESS_ENGINE_EQUIP,
-					ACCESS_MAINT_TUNNELS, ACCESS_MECH_ENGINE, ACCESS_NETWORK, ACCESS_RESEARCH)
-	minimal_access = list(ACCESS_TCOMSAT, ACCESS_TCOM_ADMIN, ACCESS_TECH_STORAGE, ACCESS_RC_ANNOUNCE, ACCESS_CONSTRUCTION, ACCESS_MECH_ENGINE, ACCESS_NETWORK, ACCESS_RESEARCH, ACCESS_TOX, ACCESS_MINISAT)
+	added_access = list(ACCESS_ENGINE_EQUIP, ACCESS_MAINT_TUNNELS)
```
Comment (adamsong) [inline yogstation/code/modules/jobs/job_types/network_admin.dm]:
```suggestion
	added_access = list(ACCESS_ENGINE, ACCESS_ENGINE_EQUIP, ACCESS_MAINT_TUNNELS)
```

File: yogstation/code/modules/jobs/job_types/network_admin.dm [hunk 9]
```diff
@@ -14,18 +14,17 @@
...
+	base_access = list(ACCESS_TCOMSAT, ACCESS_TCOM_ADMIN, ACCESS_TECH_STORAGE, ACCESS_RC_ANNOUNCE, ACCESS_CONSTRUCTION, ACCESS_MECH_ENGINE, ACCESS_NETWORK, ACCESS_RESEARCH, ACCESS_MINISAT)
```
Comment (adamsong) [inline yogstation/code/modules/jobs/job_types/network_admin.dm:18]:
```suggestion
	base_access = list(ACCESS_TCOMSAT, ACCESS_TCOM_ADMIN, ACCESS_TECH_STORAGE, ACCESS_RC_ANNOUNCE, ACCESS_CONSTRUCTION, ACCESS_MECH_ENGINE, ACCESS_NETWORK, ACCESS_RESEARCH, ACCESS_TOX, ACCESS_MINISAT)
```

**Review by adamsong — CHANGES_REQUESTED** [review]

File: yogstation/code/modules/jobs/job_types/network_admin.dm (hunk 9 above)
Comment (redmoogle) [reply yogstation/code/modules/jobs/job_types/network_admin.dm:18]: Network admins arent support to have toxins access

File: code/modules/jobs/job_types/roboticist.dm (hunk 5 above)
Comment (redmoogle) [reply code/modules/jobs/job_types/roboticist.dm:23]: Merged from yogstation folder

**Review by JamieD1 — APPROVED** [review]

**Pull request merged**